vyper/parser/context.py

```diff
@@ -144,10 +144,17 @@ def is_valid_varname(self, name, pos):
                 raise VariableDeclarationException(f"Duplicate variable name: {name}", name)
         return True
 
+    def _mangle(self, name):
+        return '#internal' + name
```

Comment:
```suggestion
        # Using comment syntax in the variable mangling at this step will
        # ensure it is not possible to use an internal variable in source code
        return '#internal' + name
```

Reply:
It's a little magical how you fixed the issue, I just wanted a better comment to that effect.

Reply:
how about, "ensure it is not possible to use an internal variable in source code because source code identifiers cannot start with `#`".

Reply:
sure

Reply:
@fubuloubu see 781aa88dd18cf8e2678da9b7bd2d941816f91f4f